engine/src/main/java/org/terasology/rendering/particles/internal/ParticleSystemManager.java

```diff
@@ -0,0 +1,213 @@
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Physics physics;
+
+    private ParticleRenderer particleRenderer;
+    private ParticleUpdater particleUpdater;
+
+    private BiMap<Class<Component>, GeneratorFunction> registeredGeneratorFunctions = HashBiMap.create();
+    private BiMap<Class<Component>, AffectorFunction> registeredAffectorFunctions = HashBiMap.create();
+
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemActivated(OnActivatedComponent event, EntityRef entity, ParticleSystemComponent particleSystemComponent) {
+        particleSystemComponent.entityRef = entity;
+
+        particleUpdater.register(entity);
+        particleUpdater.updateStateData(entity, registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemDeactivated(BeforeDeactivateComponent event, EntityRef entity) {
+        particleUpdater.dispose(entity);
+    }
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemChanged(ParticleSystemUpdateEvent event, EntityRef entity) {
+        particleUpdater.updateStateData(entity, registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+    @ReceiveEvent(components = {ParticleEmitterComponent.class})
+    public void onEmitterActivated(OnActivatedComponent event, EntityRef entity, ParticleEmitterComponent particleEmitterComponent) {
+        particleEmitterComponent.entityRef = entity;
+    }
+
+    @ReceiveEvent(components = {ParticleEmitterComponent.class})
+    public void onEmitterChanged(ParticleSystemUpdateEvent event, EntityRef entity) {
+        particleUpdater.updateStateData(entity.getOwner(), registeredAffectorFunctions, registeredGeneratorFunctions);
+    }
+
+
+    public void initialise() {
+        particleRenderer = ParticleRenderer.create(logger);
+        particleUpdater = ParticleUpdater.create(physics, logger);
+
+        registerGeneratorFunction(new ColorRangeGeneratorFunction());
+        registerGeneratorFunction(new EnergyRangeGeneratorFunction());
+        registerGeneratorFunction(new PositionRangeGeneratorFunction());
+        registerGeneratorFunction(new ScaleRangeGeneratorFunction());
+        registerGeneratorFunction(new VelocityRangeGeneratorFunction());
+        registerGeneratorFunction(new TextureOffsetGeneratorFunction());
+
+        registerAffectorFunction(new DampingAffectorFunction());
+        registerAffectorFunction(new EnergyColorAffectorFunction());
+        registerAffectorFunction(new EnergyScaleAffectorFunction());
+        registerAffectorFunction(new ForceAffectorFunction());
+        registerAffectorFunction(new PointForceAffectorFunction());
+        registerAffectorFunction(new ForwardEulerAffectorFunction());
+        registerAffectorFunction(new TurbulenceAffectorFunction());
```

Comment:
Maybe go the full distance and don't have example affectors and generators in engine but only in a new ParticleTutorialModule. That would match our setup with other things like world gen. 
On the other hand, there might be some very basic affectors/generators that shoul be available per default(?)